src/components/LanguageSwitcher/LanguageSwitcher.js

```diff
@@ -19,17 +24,23 @@ const LanguageSwitcher = () => {
    */
   const handleLanguageSelection = (event) => {
     const targetElement = event.currentTarget;
-    const languageCode = targetElement.getAttribute("data-lang-code");
-    if (languageCode) {
-      dispatch(changeLanguage(languageCode));
+    const locale = targetElement.getAttribute("data-lang-code");
+    if (locale) {
+      dispatch(changeLanguage(locale));
```

Comment:
Is changeLanguage action/reducer still needed?

Reply:
this reducer is need as we are managing the language when user is not logged in